worker/jobs/bulkUpsertSamplesJob.js

```diff
@@ -17,6 +17,13 @@ const cacheSampleModel = require('../../cache/models/samples');
 const publisher = require('../../realtime/redisPublisher');
 
 module.exports = (job, done) => {
+  if (global.gc) {
+    global.gc();
+  } else {
+    logger.info('Garbage Collection unavailable. Pass --expose-gc when ' +
+      'starting the worker');
+  }
```

Comment:
This fixes the memory issue and this will not effect the response time too. So, this is one solution but may not be the most optimal. I want to merge #820, before deploying this in prod.